src/components/MasterDetails.js

```diff
@@ -137,6 +137,31 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   }
   state: MasterDetailsState;
 
+  componentWillReceiveProps(newProps: MasterDetailsProps) {
+    const isRowPresent = newProps.rows.find((row) => {
+      return (row.id === this.state.detailsRowId);
+    });
```

Comment:
We actually need to check all of the IDs in this.state.selectedRows to see if they're in the no longer in the list of rows, and remove them if they're not, regardless of multi-select.

Once the selectedRows array is up to date, we can look at the multiSelect flag: if multiSelect is true and selectedRows has no elements, we would want to add the ID of the first item from props.rows (if props.rows is an empty list, then there's nothing we can do to ensure there's a selection—in this case selectedRows would still be empty.

Once we've updated selectedRows, we can figure out which ID to use for the detailsRowId. If selectedRows is empty, then detailsRowId will be the empty string. If it's not empty, detailsRowId will be the last item in selectedRows. (Note that if the previous row being shown in the details component is still in the list of rows, then it will still end up being detailsRowId and the details view won't change unless the row's data has changed.)

Finally, after all of that, we can update the state with the new list of selectedRows and the new detailsRowId.